### ISSUE TITLE
Redshift generates text fields that are too small

### ISSUE THREAD

**Issue by boringusername99** [transferwise/pipelinewise#423]
When syncing from MySQL to Redshift, a table with varchar >256 leads to a varchar(256) in Redshift, leading to errors in Redshift like:

`Value too long for type character varying(256)`

For example, my MySQL table has a varchar(500) but pipelinewise (really the underlying Singer target) only generates a varchar(256).

I was able to workaround this by manually changing the size to varchar(500).

**Comment by JasonSanDiego** [comment]
I did some digging around and it looks like the Redshift target is fully custom to Pipelinewise. It looks totally different from the official Singer Redshift target. I also saw that the default length of characters is supposed to be 10K chars. So, I'm not sure why all of my text fields are 256 by default, and not expanding as needed.

Perhaps, the MySQL tap isn't generating the right properties file? How custom is that tap relative to the official Singer one?

**Comment by koszti** [comment]
You're right, target-redshift is written from scratch for PPW but as far as I know there is no official open source target-redshift connector by Singer. There is a target-redshift connector written by DataMill: https://github.com/datamill-co/target-redshift , but there are certain differences in the two implementations, for example denesting objects, or adding new columns.

Regarding your issue I think this seems to be a bug in the mysql-to-redshift fastsync. Fastsync and target-redshift column mappings should be in sync but seems like it's not the case. Fixing it looks relatively straightforward:
* [mysql-to-redshift fastsync columns mapping](https://github.com/transferwise/pipelinewise-target-redshift/blob/a1416e7aad01ee3b6915f95e7ed0e392b5f44dfd/target_redshift/db_sync.py#L50)
* [target-redshift column type](https://github.com/transferwise/pipelinewise-target-redshift/blob/master/target_redshift/db_sync.py#L50)

---
Fast Sync is the component that we run at the first time when running the tap making the initial syncs finishing fater. More info in the [Fast Sync Documentation](https://transferwise.github.io/pipelinewise/user_guide/fastsync.html) :

_"Fast Sync is one of the Replication Methods that is functionally identical to Full Table replication but Fast Sync is bypassing the Singer Specification for optimised performance. Primary use case of Fast Sync is initial sync or to resync large tables with hundreds of millions of rows where singer component would usually be running for long hours or sometimes for days."_
